source/blog/2018-05-24-podman-userns.md

```diff
@@ -0,0 +1,157 @@
+---
+title: User namespaces support in Podman
+author: giuseppe
+date: 2018-05-24 00:00:00 UTC
+layout: post
+comments: false
+categories:
+- Blog
+---
+
+## User namespaces support in Podman
+
+We recently added support for user namespaces to
+[Podman](https://github.com/projectatomic/libpod). This has some major
+benefits for security and added flexibility when running
+containers. Allows processes to have privileges inside of the
+container, but no privileges if they escape the container.
+
+READMORE
+
+[User namespaces](https://lwn.net/Articles/532593/) are a kernel
+mechanism for allowing a process to see a different uids/gids mapping
+than it has in reality.  It permits mapping multiple ranges of
+uids/gids from the host to completely different values into the
+container process.
+
+This enables things like having `root` in the container but the
+process actually runs with a UID that is not root outside of the
+container. This means if the processes in the container manages to
+break out of the container, they will be treated as normal
+non-privileged processes on the host.  Of course other security
```

Comment:
Is having zero as your (apparent) uid really significant?  From [`capabilities(7)`][1]:

> For the purpose of performing permission checks, traditional UNIX implementations distinguish two categories of processes: privileged processes (whose effective user ID is 0, referred to as superuser or root), and unprivileged processes (whose effective UID is nonzero).  Privileged processes bypass all kernel permission checks, while unprivileged processes are subject to full permission checking based on the process's credentials (usually: effective UID, effective GID, and supplementary group list).
>
> Starting with kernel 2.2, Linux divides the privileges traditionally associated with superuser into distinct units, known as capabilities, which can be independently enabled and disabled.  Capabilities are a per-thread attribute.

I think the argument for privileged users to use user-namespaced containers is isolation from host filesystem access (e.g. `/etc/shadow` being readable and writable by uid 0).  Isolation from other things (e.g. binding to low ports in the host network namespace) is covered independently by capabilities.

[1]: http://man7.org/linux/man-pages/man7/capabilities.7.html

Reply:
thanks, that is what I wanted to imply.  UserNS restricts the container further from host filesystem resources that it could access.  I'll make it clearer